lib/shopify_api/bulk/query.ex

```diff
@@ -169,4 +180,60 @@ defmodule ShopifyAPI.Bulk.Query do
       _ -> poll(token, bulk_query_id, polling_rate, max_poll, depth + 1)
     end
   end
+
+  defp httpoison_streamed_get(url) do
+    Stream.resource(
+      fn ->
+        HTTPoison.get!(url, %{}, stream_to: self(), async: :once)
+      end,
+      fn %HTTPoison.AsyncResponse{id: id} = resp ->
+        receive do
+          %HTTPoison.AsyncStatus{id: ^id, code: _code} ->
+            HTTPoison.stream_next(resp)
+            {[], resp}
+
+          %HTTPoison.AsyncHeaders{id: ^id, headers: _headers} ->
+            HTTPoison.stream_next(resp)
+            {[], resp}
+
+          %HTTPoison.AsyncChunk{id: ^id, chunk: chunk} ->
+            HTTPoison.stream_next(resp)
+            {[chunk], resp}
+
+          %HTTPoison.AsyncEnd{id: ^id} ->
+            {:halt, resp}
+        after
+          @stream_http_timeout -> raise "receive timeout"
+        end
+      end,
+      fn _resp -> :ok end
+    )
+  end
+
+  defp transform_chunks_to_jsonl(element, acc) do
+    # Our http chunks are not guaranteed to be on the line break, this splits up our
+    # returned chunks by line break.
+    element = acc <> element
+
+    cond do
```

Comment:
```suggestion
    case acc <> element do
```

Reply:
I guess it's not much of an advantage since then you can't do the things you're currently using `cond` for, right...

Would at least maybe be good to give a more semantically appropriate name to the combined `acc <> element`, maybe `chunk`?

Reply:
well... incoming element is `chunk` here... acc <> chunk is .... new acc or ?

Reply:
Taking an incoming param, doing something to it, and reassigning it is a pretty common pattern.